dulwich/tests/test_index.py

```diff
@@ -369,7 +369,10 @@ def test_nonempty(self):
                 sorted(os.listdir(os.path.join(repo.path, 'c'))))
 
     def test_norewrite(self):
-        sync = getattr(os, 'sync', lambda: os.system('sync'))
+        if sys.platform != 'win32':
```

Comment:
Ignoring these is going to lead to spurious errors. Perhaps we can just stick to calling os.sync when it is available and skipping this test otherwise (e.g. using @skipIf)? os.sync is available in python3 on windows too AFAIK.